Sources/SwifterSwift/Foundation/NSRegularExpressionExtensions.swift

```diff
@@ -0,0 +1,146 @@
+//
+//  NSRegularExpressionExtensions.swift
+//  SwifterSwift
+//
+//  Created by Guy Kogus on 09/10/2019.
+//  Copyright © 2019 SwifterSwift
+//
+
+#if canImport(Foundation)
+import Foundation
+
+public extension NSRegularExpression {
+
+    /// Enumerates the string allowing the Block to handle each regular expression match.
+    ///
+    /// - Parameters:
+    ///   - string: The string.
+    ///   - options: The matching options to report. See `NSRegularExpression.MatchingOptions` for the supported values.
+    ///   - range: The range of the string to test.
+    ///   - block: The Block enumerates the matches of the regular expression in the string.
+    ///     The block takes three arguments and returns `Void`:
+    ///   - result:
+    ///     An `NSTextCheckingResult` specifying the match. This result gives the overall matched range via its `range` property, and the range of each individual capture group via its `range(at:)` method. The range {NSNotFound, 0} is returned if one of the capture groups did not participate in this particular match.
+    ///   - flags:
+    ///     The current state of the matching progress. See `NSRegularExpression.MatchingFlags` for the possible values.
+    ///   - stop:
+    ///     A reference to a Boolean value. The Block can set the value to true to stop further processing of the array. The stop argument is an out-only argument. You should only ever set this Boolean to true within the Block.
+    func enumerateMatches(in string: String,
+                          options: MatchingOptions = [],
+                          range: Range<String.Index>,
+                          using block: (_ result: NSTextCheckingResult?, _ flags: MatchingFlags, _ stop: inout Bool) -> Void) {
+        enumerateMatches(in: string,
+                         options: options,
+                         range: NSRange(range, in: string)) { result, flags, stop in
+                            var shouldStop = false
```

Comment:
I don't really love it because it forces a return value, which for the most part will be `false`, instead of just letting users set the `stop` param when necessary. You can see what I mean in my unit test, which would become:
```Swift
func testEnumerateMatches() {
    var count = 0
    regularExpression.enumerateMatches(in: string,
                                       options: [],
                                       range: string.startIndex..<string.endIndex) { (result, _) in
                                        XCTAssertEqual(String(string[Range(result!.range, in: string)!]), searchString)
                                        count += 1
                                        return false
    }
    XCTAssertEqual(count, expectedMatches)

    count = 0
    let max = 2 // must be less than expectedMatches
    regularExpression.enumerateMatches(in: string,
                                       options: [],
                                       range: string.startIndex..<string.endIndex) { (result, _) in
                                        XCTAssertEqual(String(string[Range(result!.range, in: string)!]), searchString)
                                        count += 1
                                        return count >= max
    }
    XCTAssertEqual(count, max)
}
```

Reply:
I'm fine with it, was just a question. I see the point, makes sense :))
It just that I personally prefer return, because from the caller propective is always more clear when return (at least to me).